src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3604,7 +3604,10 @@ trait Types
       if (sym.isAliasType && sameLength(sym.info.typeParams, args) && !sym.lockOK)
         throw new RecoverableCyclicReference(sym)
 
-      TypeRef(pre, sym, args)
+      if ((args eq Nil) && (pre eq NoPrefix))
+        sym.tpeHK // opt lean on TypeSymbol#tyconCache, rather than interning a type ref.
```

Comment:
IIRC, we can't (easily) do that because `tpeHK` calls that `TypeRef.apply`, so we'd loop.